src/targets/shared/web/ResizeContainer.tsx

```diff
@@ -71,34 +73,48 @@ const ResizeContainer = React.memo(function ResizeContainer(props: ResizeContain
     ...restSpread
   } = props
 
-  const containerRef = useRef<HTMLDivElement>()
+  const containerRef = React.useRef<HTMLDivElement | null>(null)
   // onGotPointerCaptureLegacy is a fake event used by non-web targets to simulate poinzter capture
-  const [{ onGotPointerCaptureLegacy, ...events }, setEvents] = useState<DomEventHandlers>({} as DomEventHandlers)
-  const [bind, size, forceResize] = useMeasure({ scroll: true, debounce: { scroll: 50, resize: 0 }, ...resize })
+  const [{ onGotPointerCaptureLegacy, ...events }, setEvents] = React.useState<DomEventHandlers>({} as DomEventHandlers)
+  const [useMeasureRef, size, forceResize] = useMeasure({
+    scroll: true,
+    debounce: { scroll: 50, resize: 0 },
+    ...resize,
+  })
 
   // Flag view ready once it's been measured out
-  const readyFlag = useRef(false)
-  const ready = useMemo(() => (readyFlag.current = readyFlag.current || (!!size.width && !!size.height)), [size])
-  const state = useMemo(() => ({ size, forceResize, setEvents, container: containerRef.current as HTMLDivElement }), [
-    forceResize,
-    size,
-  ])
+  const readyFlag = React.useRef(false)
+  const ready = React.useMemo(() => (readyFlag.current = readyFlag.current || (!!size.width && !!size.height)), [size])
+  const state = React.useMemo(
+    () => ({ size, forceResize, setEvents, container: containerRef.current as HTMLDivElement }),
+    [forceResize, size]
+  )
+
+  const memoStyle = React.useMemo(
+    function memo() {
+      return { ...defaultStyles, ...style }
+    },
+    [style]
+  )
 
   // Allow Gatsby, Next and other server side apps to run. Will output styles to reduce flickering.
-  if (typeof window === 'undefined')
+  if (typeof window === 'undefined') {
     return (
-      <div style={{ ...defaultStyles, ...style }} {...restSpread}>
+      <div style={memoStyle} {...restSpread}>
         {preRender}
       </div>
     )
+  }
 
   // Render the canvas into the dom
   return (
-    <div ref={mergeRefs([bind, containerRef])} style={{ ...defaultStyles, ...style }} {...events} {...restSpread}>
+    <div ref={mergeRefs([useMeasureRef, containerRef])} style={memoStyle} {...events} {...restSpread}>
```

Comment:
here is type inference fixed for primitive ref prop

Reply:
i dont understand this part:

```jsx
const memoStyle = React.useMemo(
    function memo() {
      return { ...defaultStyles, ...style }
    },
    [style]
  )
```

it returns a function for some reason?

Reply:
but if you think it's necessary, then i prefer

const memoStyle = React.useMemo(() => ({ ...defaultStyles, ...style }), [style])

Reply:
for readability it is not great, but for performance it is better. less GC - faster perf.
I get used to named functions, better for debugging and performance audit. 
I use eslint-plugin-react-perf, and it highlights all common with passing new array/object to the component/primitive.

Reply:
yeah but that's too much, just not my style - i find it uncomfortable. with all the usememo one liners in the code it would blow it up 20%. we can use the memo, but imo every line counts.

Reply:
ok, I will change all memo functions to arrow.

Reply:
changed. Also I found incorrect exports with React.memo().
The way it was - components had Anonymous name in React tree in react-dev-tools. Now all of them will have correct names.